expression/util.go

```diff
@@ -166,6 +166,161 @@ func extractColumns(result []*Column, expr Expression, filter func(*Column) bool
 	return result
 }
 
+func ExtractEquivalenceColumns(result [][]interface{}, exprs []Expression) [][]interface{} {
```

Comment:
Actually, the equiv class can be maintained via the selection/join?

Reply:
Yes, and what ideas about `be maintained via the selection/join` ?

Reply:
seem selection's condition will change dynamically for PropagateConstant/Decorrelate and so on, maintaining work should catch up in these position. i'll leave it here for future work.